### PR TITLE
Enhancement of bindings for executeCommand()

### LINKED ISSUE #671: executeCommand should be exposed in Python Regions
see https://github.com/htm-community/htm.core/pull/609#discussion_r324820486 
`executeCommand` is currently available in c++, and in `htm.advanced.regions`, it should be exposed to all py regions

### PR DESCRIPTION
**Pull request by dkeeney**
From python, you can now call region.executeCommand( ) using normal calling arguments.
The first argument must be the command to execute.  The subsequent arguments are converted to strings and passed to the Region as a vector of strings.

Arguments that are strings, bool and numbers are converted to std::string as nomal.
Other structures as arguments are serialized to JSON strings.

Fixes #671

### CODE REVIEW

**Comment by dkeeney** [comment]
Included with this PR are unit tests that call the executeCommand() from python into the C++ Region TestNode.  I do not have tests that:
- calls from python into a python implemented region
- calls from C++ into a python implemented region

**Comment by dkeeney** [comment]
> calls from python into a python implemented region

Ok, I managed to add another test that does this.   This is the first time I looked at the python region side of the executeCommand feature.  It does not work like the C++ regions at all.
The engine calls 'executeMethod' on the base class.  This function looks for a method with the same name as the command and calls it with the arguments.  So, the python region implementation does not implement executeCommand( ).  It implements a function with the name of the command to execute.

**Comment by dkeeney** [comment]
> calls from C++ into a python implemented region

As I think about this I realize that I am not sure this is possible.  The C++ app would have to register a python class as the region implementation.

**Comment by breznak** [comment]
> This function looks for a method with the same name as the command and calls it with the arguments. So, the python region implementation does not implement executeCommand( ). It implements a function with the name of the command to execute.

seems to me that the py version is more flexible. If I understand what you're saying: 
- py: `executeCommand("myFn", ["arg1", "2", ...])` 
  finds method `myFn` and calls it with given arguments.
- cpp: "myFn" has to be a method of the region (say SPRegion, of SPRegion, or of SP?) and is called.

File: bindings/py/cpp_src/bindings/engine/py_Engine.cpp [hunk 1]
```diff
@@ -241,56 +242,77 @@ namespace htm_ext
...
+            .def("setParameterInt64",  &Region::setParameterInt64)
+            .def("setParameterUInt64", &Region::setParameterUInt64)
+            .def("setParameterReal32", &Region::setParameterReal32)
+            .def("setParameterReal64", &Region::setParameterReal64)
+            .def("setParameterBool",   &Region::setParameterBool)
+            .def("setParameterString", &Region::setParameterString)
+            .def("setParameterArray",  &Region::setParameterArray);
+                
+                
+        py_Region.def("executeCommand", [](Region& r, const std::string& command, py::args args)
```
Comment (breznak) [inline bindings/py/cpp_src/bindings/engine/py_Engine.cpp]: so this is a convenience wrapper for `executeCommand("command" vector<string>)` ?

File: bindings/py/tests/network_test.py [hunk 2]
```diff
@@ -301,5 +304,31 @@ def testBuiltInRegions(self):
...
+    """
+    Check to confirm that the ExecuteCommand( ) funtion works.
+    From Python calling a C++ region.
+    """
+    net = engine.Network()
+    r = net.addRegion("test", "TestNode", "")
+    
+    lst = ["list arg", 86]
+    result = r.executeCommand("HelloWorld", 42, lst)
+    self.assertTrue(result == "Hello World says: arg1=42 arg2=['list arg', 86]")
```
Comment (breznak) [inline bindings/py/tests/network_test.py]: very nice test, thanks! :+1:

**Comment by dkeeney** [comment]
> seems to me that the py version is more flexible.

yes, but C++ cannot do that.  C++ needs to know everything at compile time.

> PS: next time please try to keep the whitespace formatting to a separate commit

Sorry.  Just could not help myself.  Had to remove the tabs.

**Pull request merged**